pkg/signatures/public_key_fetcher.go

```diff
@@ -232,3 +237,42 @@ func checkIfErrorContainsCode(err error, codes ...int) bool {
 
 	return false
 }
+
+var (
+	_ oci.SignedEntity = (*localSignedEntity)(nil)
+)
+
+// localSignedEntity is an implementation of oci.SignedEntity used for fetching signatures.
+// This implementation skips fetching the manifest of the signed image, since within the image enriching, we already
+// fetched the image manifest beforehand.
+type localSignedEntity struct {
+	oci.SignedEntity
+	opts   []ociremote.Option
+	imgRef name.Reference
+	imgSHA string
+}
+
+func newLocalSignedEntity(img *storage.Image, imgRef name.Reference, opts ...ociremote.Option) *localSignedEntity {
+	imgSHA := imgUtils.GetSHA(img)
+	return &localSignedEntity{
+		opts:   opts,
+		imgRef: imgRef,
+		imgSHA: imgSHA,
+	}
+}
+
+func (s *localSignedEntity) Digest() (v1.Hash, error) {
+	return v1.NewHash(s.imgSHA)
+}
+
+func (s *localSignedEntity) Signatures() (oci.Signatures, error) {
+	h, err := s.Digest()
+	if err != nil {
+		return nil, err
+	}
+	// The name reference of the signature to fetch is going to be:
+	// <registry>/<repository>@<digest>.sig
+	// This is being kept in line with:
+	// https://github.com/sigstore/cosign/blob/65eb28af970d133adeefdc6c48d6e9304dd8cc3a/pkg/oci/remote/remote.go#L87
+	return ociremote.Signatures(s.imgRef.Context().Tag(fmt.Sprint(h.Algorithm, "-", h.Hex, ".sig")), s.opts...)
```

Comment:
I think the suffix is an option in the cosign library. Maybe we should also have it as an option (which defaults to `.sig`) here

Reply:
When we initially added the signature feature, we decided to not include this for the initial version, only if there was actual customer requests to include that.

So, since then no one has really asked for it, I'd say we don't introduce it now but if we get feedback that someone actually requires it (in my mind it would have been something to add when we add support for keyless signature verification via rekor).